teapots/choreographer-30fps/src/main/cpp/CMakeLists.txt

```diff
@@ -15,38 +15,39 @@
 #
 
 cmake_minimum_required(VERSION 3.4.1)
+project(ChoreographerNativeActivity LANGUAGES C CXX)
 
-# build native_app_glue as a static lib
-add_library(native_app_glue STATIC
-    ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
+# set up common compile options
+set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wall -Werror -fno-exceptions -fno-rtti")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
-
-# build the ndk-helper library
-set(ndk_helper_dir ../../../../common/ndk_helper)
-add_subdirectory(${ndk_helper_dir} ndk_helper)
-
-# Export ANativeActivity_onCreate(), 
-# Refer to: https://github.com/android-ndk/ndk/issues/381.
-set(CMAKE_SHARED_LINKER_FLAGS
-    "${CMAKE_SHARED_LINKER_FLAGS} -u ANativeActivity_onCreate")
+# build the ndk-helper library into a common directory
+# visible to all 3 projects avoiding duplicate builds.
+get_filename_component(ndkHelperSrc ../../../../common/ndk_helper ABSOLUTE)
+get_filename_component(ndkHelperBin ${ndkHelperSrc}/../ndkHelperBin ABSOLUTE)
+add_subdirectory(${ndkHelperSrc} ${ndkHelperBin})
 
 # now build app's shared lib
-add_library(ChoreographerNativeActivity SHARED
+add_library(${PROJECT_NAME}
+  SHARED
     ChoreographerNativeActivity.cpp
-    TeapotRenderer.cpp)
+    TeapotRenderer.cpp
+)
 
-target_include_directories(ChoreographerNativeActivity PRIVATE
-    ${ANDROID_NDK}/sources/android/cpufeatures
-    ${ANDROID_NDK}/sources/android/native_app_glue
-    ${ndk_helper_dir})
+set_target_properties(${PROJECT_NAME}
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
+
+# Export ANativeActivity_onCreate(),
+# Refer to: https://github.com/android-ndk/ndk/issues/381.
+set_target_properties(${PROJECT_NAME}
+  PROPERTIES  LINK_FLAGS "-u ANativeActivity_onCreate")
 
 # add lib dependencies
-target_link_libraries(ChoreographerNativeActivity
```

Comment:
Aren't these needed by this project? (same goes for all the other cases where this changed)

Reply:
[project](https://cmake.org/cmake/help/v3.0/command/project.html)(ChoreographerNativeActivity LANGUAGES C CXX)
set ${PROJET_NAME} to be the first parameter, and "The top-level CMakeLists.txt file for a project must contain a literal, direct call to the project() command; loading one through the include() command is not sufficient. If no such call exists CMake will implicitly add one to the top that enables the default languages (C and CXX)."